src/CurlRequest.php

```diff
@@ -25,9 +25,8 @@ class CurlRequest
      */
     public function with(callable $callback)
     {
-        $clone = clone $this;
```

Comment:
This removes immutability, which is required to keep consistent with how PSR handles the HTTP request and response objects (and also Slim)